sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaMetrics.java

```diff
@@ -146,14 +154,31 @@ private void recordRpcLatencyMetrics() {
       }
     }
 
-    private void recordBacklogBytes() {
+    private void recordBacklogBytesInternal() {
       for (Map.Entry<String, Long> backlogs : perTopicPartitionBacklogs().entrySet()) {
         Gauge gauge =
             KafkaSinkMetrics.createBacklogGauge(MetricName.named("KafkaSink", backlogs.getKey()));
         gauge.set(backlogs.getValue());
       }
     }
 
+    /**
+     * This is for recording backlog bytes on the current thread.
+     *
+     * @param topicName topicName
+     * @param partitionId partitionId for the topic Only included in the metric key if
+     *     'supportsMetricsDeletion' is enabled.
+     * @param backlogBytes backlog for the topic Only included in the metric key if
+     *     'supportsMetricsDeletion' is enabled.
+     */
+    @Override
+    public void recordBacklogBytes(String topicName, int partitionId, long backlogBytes) {
+      Gauge perPartion =
+          Metrics.gauge(
```

Comment:
ie can we create the gauge here with the getPerWorkerGauge method? and then ensure that those metrics have the per-worker metadata populated appropriately?

Reply:
We create it as a regular gauge, and then add per worker metadata to differenciate between the two (with the metric labels). If we add a new api here, it would then be added to any other derived metrics, (and then we'd likely want per worker other sorts of metrics.) Keeping the original set (gauge, hist, counter, etc) and then adding metadata for runner aggregation might be a cleaner approach?